actors/builtin/miner/miner_actor.go

```diff
@@ -1140,7 +1158,8 @@ func processMissingPoStFaults(rt Runtime, st *State, store adt.Store, deadlines
 	builtin.RequireNoErr(rt, err, exitcode.ErrIllegalState, "failed to load failed recovery sectors")
 
 	// Unlock sector penalty for all undeclared faults.
-	penalty, err := unlockPenalty(st, store, currEpoch, append(detectedFaultSectors, failedRecoverySectors...), pledgePenaltyForSectorUndeclaredFault)
+	penaltyCalculator := computeUndeclaredFaultPenalty(epochReward, currentTotalPower)
```

Comment:
Pass the penalty calculator as the parameter, instead of the two values. You can move and re-use the one currently at 1066.